src/wrappers/themis/react-native-themis/ios/RCTThemis.m

```diff
@@ -120,59 +126,57 @@ + (NSData*)dataDeserialize:(NSArray*) data
 
   NSArray  *privateKey = [RCTThemis dataSerialize: keypair.privateKey];
   NSArray   *publicKey = [RCTThemis dataSerialize: keypair.publicKey];
-  
+
   NSDictionary *dictionary = @{
-       @"private" : privateKey,
-       @"public"  : publicKey
+    @"private" : privateKey,
+    @"public"  : publicKey
   };
-  callback(@[dictionary]);
+  successCallback(@[dictionary]);
 }
 
 
 RCT_EXPORT_METHOD(symmetricKey:(RCTResponseSenderBlock)callback)
 {
   NSData *symmetricKey = TSGenerateSymmetricKey();
-    NSArray *masterKey = [RCTThemis dataSerialize: symmetricKey];
+  NSArray *masterKey = [RCTThemis dataSerialize: symmetricKey];
   callback(@[masterKey]);
 }
 
-- (TSCellSeal *)newSealMode: (NSArray*) symmetricKey
+- (TSCellSeal *)newSealMode: (NSArray*) symmetricKey error:(NSError**) error
 {
-  @try {
-    NSData *masterKey = [RCTThemis dataDeserialize: symmetricKey];
-    TSCellSeal *cell  = [[TSCellSeal alloc] initWithKey:masterKey];
-    return cell;
+  NSData *masterKey = [RCTThemis dataDeserialize: symmetricKey error:error];
+  if (masterKey == nil) {
+    return nil;
   }
-  @catch (NSException *e) {
-    @throw e; // rethrow to catch in final function
-  }
-
+  TSCellSeal *cell  = [[TSCellSeal alloc] initWithKey:masterKey];
+  return cell;
 }
 
 
-
-
-
 RCT_EXPORT_METHOD(secureCellSealWithSymmetricKeyEncrypt:(NSArray*) symmetricKey
                   plaintext: (NSString*)plaintext
                   context: (NSString*)context
                   successCallback: (RCTResponseSenderBlock)successCallback
                   errorCallback: (RCTResponseErrorBlock)errorCallback)
 {
-
   TSCellSeal *cell;
-  @try {
-    cell  = [self newSealMode:symmetricKey];
+  NSError *error;
+
+  cell = [self newSealMode:symmetricKey error:&error];
+  if (cell == nil) {
+    errorCallback(error);
+    return;
   }
-  @catch (NSException *e) {
-    NSError* error = SCERROR(BYTEOVERFLOW, e.reason);
+
+  NSData *plaintextBinary  = [plaintext dataUsingEncoding:NSUTF8StringEncoding];
+  NSData *contextBinary  = [context dataUsingEncoding:NSUTF8StringEncoding];
+  NSData  *encrypted = [cell  encrypt:plaintextBinary
+                              context:contextBinary
+                                error:&error];
+  if (encrypted == nil) {
```

Comment:
No. Because we can't call encrypt method with nil values because we filtered parameters for empty strings. Even if the method returns nil it means an error occurs. Tests confirmed. I will add the tests with the next PR.

Reply:
see above